lib/motors/vector_controller.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "core/types.hpp"
+
+namespace hyped::motors {
+class VectorController {
+ private:
+  core::Float Ia;
+  core::Float Ib;
+  core::Float Ic;
+  core::Float rotorVelocity;
+  core::Float Id;
+  core::Float Iq;
+  core::Float Vd;
+  core::Float Vq;
+  core::Float IdRef;
+  core::Float IqRef;
+  core::Float Kp;
+  core::Float Ki;
+  core::Float theta;
+
+ public:
+  VectorController();
+
+  ~VectorController();
+
+  void measureCurrentsAndVelocity(core::Float ia, core::Float ib, core::Float ic, core::Float rotorVelocity);
+  void convertTo2Axis();
+  void inverse2Axis();
+  void piControl();
+  void inverseParkTransform();
```

Comment:
most of these should be private, and used by public methods

Pass in the new currents and velocity and do the rest of it internally unless these methods definitely need to be called elsewhere